Add 4.4-dev6 article (#971)
godotengine/godot-website: collections/_article/dev-snapshot-godot-4-4-dev-6.md

---
title: "Dev snapshot: Godot 4.4 dev 6"
excerpt: "Featuring: Malware! (does not contain malware)"
categories: ["pre-release"]
author: Thaddeus Crews
image: /storage/blog/covers/dev-snapshot-godot-4-4-dev-6.webp
image_caption_title: "Malware"

> image_caption_description: "A game by OddGames"

Reviewer: ```suggestion
image_caption_description: "A game by Odd Games"
```

> date: 2024-12-04 12:00:00

Reviewer: ```suggestion
date: 2024-12-05 17:00:00
```

---

> Safety goggles on, people! Features and Quality-of-Life are being integrated at blinding speeds; precautions must be taken to view them head-on. Once secure, we can dive into what's shaping up to be our final development snapshot before 4.4-beta builds roll out!

Reviewer: Since we'll have dev7 as discussed on RC:
```suggestion
Safety goggles on, people! Features and Quality-of-Life are being integrated at blinding speeds; precautions must be taken to view them head-on. Once secure, we can dive into what's shaping up to be one of the final development snapshots before 4.4-beta builds roll out!
```

Could rephrase further to avoid people misreading it as what it said initially.

> 
> Many of the changes in this release are bug fixes that will be backported to Godot 4.3 and released in 4.3.1! So please
> test this release well so we can be confident with the changes and release 4.3.1 with them as soon as possible.

Reviewer: I think we can drop this by now, it was mostly meant for the first dev snapshots as usually we release the first patch release around the same time as dev1 or dev2 with similar (bugfix) changes.
```suggestion
```

Keep in mind that while we try to make sure each dev snapshot is stable enough for general testing, this is by
definition a pre-release piece of software. Be sure to make frequent backups, or use a version control system such as
Git, to preserve your projects in case of corruption or data loss.

[Jump to the **Downloads** section](#downloads), and give it a spin right now, or continue reading to learn more about improvements in this release. You can also try the [**Web editor**](https://editor.godotengine.org/releases/4.4.dev6/), [**XR editor**](https://www.meta.com/experiences/godot-game-engine/7713660705416473/), or the **Android editor** for this release (join the [Android editor testing group](https://groups.google.com/g/godot-testers) to get access to pre-release builds).

-----

> *The cover illustration is from* [**Malware**](https://store.steampowered.com/app/3019370/Malware/), *an installation simulator where players must navigate a minefield of circa-1999 maleware! It is developed by [OddGames](https://store.steampowered.com/search/?developer=Odd%20Games). You can purchase the game [on Steam](https://store.steampowered.com/app/3019370/Malware/).*

Reviewer 2: ```suggestion
*The cover illustration is from* [**Malware**](https://store.steampowered.com/app/3019370/Malware/), *an installation simulator where players must navigate a minefield of circa-1999 malware! It is developed by [Odd Games](https://store.steampowered.com/search/?developer=Odd%20Games). You can purchase the game [on Steam](https://store.steampowered.com/app/3019370/Malware/).*
```

although I was tempted to suggest:

```suggestion
*The cover illustration is from* [**Malware**](https://store.steampowered.com/app/3019370/Malware/), *an installation simulator where players must navigate a minefield of circa-1999 menswear! It is developed by [OddGames](https://store.steampowered.com/search/?developer=Odd%20Games). You can purchase the game [on Steam](https://store.steampowered.com/app/3019370/Malware/).*
```

## Highlights

In case you missed them, see the [4.4 dev 1](/article/dev-snapshot-godot-4-4-dev-1/), [4.4 dev 2](/article/dev-snapshot-godot-4-4-dev-2/),
[4.4 dev 3](/article/dev-snapshot-godot-4-4-dev-3/), [4.4 dev 4](/article/dev-snapshot-godot-4-4-dev-4/), and [4.4 dev 5](/article/dev-snapshot-godot-4-4-dev-5/) release notes for an overview of

> some key features which were already in that snapshot, and are therefore still available for testing in dev 6.

Reviewer 2: ```suggestion
some key features which were already in those snapshots, and are therefore still available for testing in dev 6.
```

Here are highlights of a few new features in dev 6 that you might find particularly exciting!

### Camera3D preview in inspector

Words words words

<video autoplay loop muted playsinline>
  <source src="/storage/blog/dev-snapshot-godot-4-4-dev-6/camera3d-inspector.mp4?1" type="video/mp4">
</video>

### CollisionShape3D debug color customization

Words words words

![Examples of custom debug colors](/storage/blog/dev-snapshot-godot-4-4-dev-6/debug-color.webp)

> ### And more!

Reviewer 3: I'm plugging my own feature here, but I think it warrants a preview. Feel free to disagree.

````suggestion
### Temporary file and directory utilities

Tool creators, here's something that could be interesting for you. A brand new API let's you create and dispose of temporary files. Handy if you need to write content on disk, but without worrying about polluting user data. These temporary files and directories even dispose of themselves after use by default. For more information, see ([GH-98397](https://github.com/godotengine/godot/pull/98397)).

### And more!
````

- Audio: Fix AudioStreamPlayer2D/3D's `get_playback_position` returning `0` right after calling `seek` ([GH-99759](https://github.com/godotengine/godot/pull/99759)).
- C#: Preserve no-hint behavior for unmarshallable generics in dictionaries ([GH-99485](https://github.com/godotengine/godot/pull/99485)).
- Core: Unix: Don't create world-writable files when safe save is enabled ([GH-96399](https://github.com/godotengine/godot/pull/96399)).
- Core: Suggest using OS.has_feature instead of the engine architecture name for bitness ([GH-99465](https://github.com/godotengine/godot/pull/99465)).
- Core: Convert line breaks to `\n` and strip line break from the end of string returned by `OS::read_string_from_stdin`/`OS::get_stdin_string` ([GH-99542](https://github.com/godotengine/godot/pull/99542)).
- Core: Fix `OS::has_feature()` skipping custom features ([GH-99864](https://github.com/godotengine/godot/pull/99864)).
- Core: Add temp utilities (alias `OS::get_temp_dir()`, `FileAccess::create_temp()`, and `DirAccess::create_temp()`) ([GH-98397](https://github.com/godotengine/godot/pull/98397)).

> - Documentation: Docs: Standardize names and terms for renderers ([GH-98744](https://github.com/godotengine/godot/pull/98744)).

Reviewer 2: ```suggestion
- Documentation: Standardize names and terms for renderers ([GH-98744](https://github.com/godotengine/godot/pull/98744)).
```

- Documentation: Remove uses of implicit static typing from the documentation ([GH-99924](https://github.com/godotengine/godot/pull/99924)).
- Editor: Extend Curve to allow for domains outside of [0, 1] ([GH-67857](https://github.com/godotengine/godot/pull/67857)).
- Editor: Implement zooming and panning in the profiler ([GH-76055](https://github.com/godotengine/godot/pull/76055)).
- Editor: Fix jumping to editor help does not scroll correctly sometimes ([GH-96449](https://github.com/godotengine/godot/pull/96449)).
- Editor: Show file names in remove files confirmation dialog ([GH-98539](https://github.com/godotengine/godot/pull/98539)).
- Editor: Add editor setting to stop the bottom panel from switching to the Stack Trace ([GH-98657](https://github.com/godotengine/godot/pull/98657)).
- Editor: Don't tint editor bottom panel icons when hovered or pressed ([GH-98765](https://github.com/godotengine/godot/pull/98765)).
- Editor: Add copy button to toast notification ([GH-98778](https://github.com/godotengine/godot/pull/98778)).
- Export: Ensure excluded GDExtension files are not included in `extension_list.cfg` ([GH-97216](https://github.com/godotengine/godot/pull/97216)).

> - GDExtension: Register Engine, OS, ProjectSettings, and Time singletons in time for for `INITIZATION_LEVEL_CORE` ([GH-98862](https://github.com/godotengine/godot/pull/98862)).

Reviewer 2: ```suggestion
- GDExtension: Register Engine, OS, ProjectSettings, and Time singletons in time for `INITIZATION_LEVEL_CORE` ([GH-98862](https://github.com/godotengine/godot/pull/98862)).
```

-----

*The cover illustration is from* [**Malware**](https://store.steampowered.com/app/3019370/Malware/), *an installation simulator where players must navigate a minefield of circa-1999 malware! It is developed by [Odd Games](https://store.steampowered.com/search/?developer=Odd%20Games). You can purchase the game [on Steam](https://store.steampowered.com/app/3019370/Malware/).*

## Highlights

In case you missed them, see the [4.4 dev 1](/article/dev-snapshot-godot-4-4-dev-1/), [4.4 dev 2](/article/dev-snapshot-godot-4-4-dev-2/),
[4.4 dev 3](/article/dev-snapshot-godot-4-4-dev-3/), [4.4 dev 4](/article/dev-snapshot-godot-4-4-dev-4/), and [4.4 dev 5](/article/dev-snapshot-godot-4-4-dev-5/) release notes for an overview of
some key features which were already in those snapshots, and are therefore still available for testing in dev 6.

Here are highlights of a few new features in dev 6 that you might find particularly exciting!

### Camera3D preview in inspector

> Words words words

Reviewer 3: ```suggestion
Previously, having multiple `Camera3D` in a scene was chore to handle. It required activating the camera you wanted to preview, and then reverting it back if it wasn't the main camera.

Now, with [timothyqiu](https://github.com/timothyqiu)'s feature, every selected 3D camera show a preview inside the inspector. No switching cameras needed to preview anymore.

See [GH-90778](https://github.com/godotengine/godot/pull/90778) for more details.
```

Reviewer: Nice! I was just writing it myself too, got this:
```
Adjusting the viewshed of cameras in the editor viewport can oftentimes be tedious. You can split the 3D viewport into multiple viewports and pin the camera's view on one of them, but this reduces the size of the main viewport where you are positioning your camera.

This new feature by [Haoyu Qiu](https://github.com/timothyqiu) lets you preview the camera in the Inspector directly, so it's always available at a glance and you can keep your main 3D viewport real estate ([GH-90778](https://github.com/godotengine/godot/pull/90778)).
```

I think your version is more readable, I'll go with that.

Reviewer: I merged both as my understanding is that this isn't just about multiple cameras, but just being able to preview any camera conveniently in the main viewport.
